File: java/basic/Armstrong.java
import java.util.Scanner;

public class Armstrong {
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);
        System.out.print("Enter a number to check: ");
        int num = scn.nextInt(); // Fixed scanner variable

        // Compute the cube using Math.pow and cast the result to int
        int result = (int) Math.pow(num, 3); // Fixed type issue
        System.out.println("The cube of " + num + " is: " + result);
    }
}

File: js/basic/armstrong.js
function armStrong(num) {
    let sum = 0;
    let temp = num;
    const length = String(num).length;
    console.log("Length:", length);

    while (temp > 0) {
        let digit = temp % 10; // extract last digit
        sum += Math.pow(digit, length); // add digit^length
        temp = Math.floor(temp / 10); // remove last digit
    }

    console.log("Sum:", sum, "Original:", num);
    return sum === num;
}

console.log(armStrong(153));  // true
console.log(armStrong(9474)); // true
console.log(armStrong(123));  // false
